tests/test_ext_napoleon_docstring.py

```diff
@@ -1950,6 +1951,160 @@ def test_list_in_parameter_description(self):
         actual = str(NumpyDocstring(docstring, config))
         self.assertEqual(expected, actual)
 
+    def test_recombine_set_tokens(self):
+        tokens = (
+            ["{", "'F'", ", ", "'C'", ", ", "'N'", "}"],
+            ["{", '"F"', ", ", '"C"', ", ", '"N"', "}"],
+            ["{", "1", ", ", "2", "}"],
+        )
+        recombined_tokens = (
+            ["{'F', 'C', 'N'}"],
+            ['{"F", "C", "N"}'],
+            ["{1, 2}"],
+        )
+
+        for input_tokens, expected in zip(tokens, recombined_tokens):
+            actual = _recombine_set_tokens(input_tokens)
+            self.assertEqual(expected, actual)
+
+    def test_recombine_set_tokens_invalid(self):
+        invalid_tokens = (
+            ["{", "1", ", ", "2"],
+        )
+
+        for input_tokens in invalid_tokens:
+            with self.assertWarnsRegex(UserWarning, "invalid value set:"):
```

Comment:
@tk0miya: how do I test that invalid value sets emit a warning? Using `assertWarnsRegex` obviously doesn't work since the warning is emitted by the logger and not the `warnings` module